--- origsrc/avahi-0.6.31/avahi-gobject/AvahiCore-0.6.gir	2010-10-04 19:29:06.042185300 -0500
+++ src/avahi-0.6.31/avahi-gobject/AvahiCore-0.6.gir	2013-01-22 21:03:19.622016900 -0600
@@ -3,7 +3,7 @@
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0"
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
-  <namespace name="AvahiCore" version="0.6" shared-library="avahi-core">
+  <namespace name="AvahiCore" version="0.6" shared-library="cygavahi-core-7.dll">
     <alias name="IfIndex" c:type="AvahiIfIndex">
       <type name="gint" c:type="int"/>
     </alias>
